### PR TITLE
Add New Branch button

### LINKED ISSUE #2784: No create a new branch button
### Description

No obvious button to create a new branch

### Version

**GitHub Desktop version:** 1.0.0

**OS version:** MacOS 10.12.6

### Steps to Reproduce

1. Add a repo
2. See that there is no button to create a new branch.

### PR DESCRIPTION
**Pull request by joshaber**
Fixes #2784

- [x] Prepopulate the branch name from the filter
- [x] Show the "No results" view

screen shot 2017-10-13 at 11 17 42 am

### CODE REVIEW

**Comment by joshaber** [comment]
This is ready for review 🙏

File: app/src/ui/branches/no-branches.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        <Button
+          className="create-branch-button"
+          onClick={this.props.onCreateNewBranch}
+          type="submit"
+        >
+          {__DARWIN__ ? 'Create New Branch' : 'Create new branch'}
+        </Button>
+
+        <div className="protip">
+          ProTip! Press <kbd>{shortcut}</kbd> + <kbd>⇧</kbd> + <kbd>N</kbd> to
```
Comment (shiftkey) [inline app/src/ui/branches/no-branches.tsx]: The arrow here needs to be `Shift` here on Windows to match up with how menu items draw their shortcuts:

File: app/src/ui/branches/branch-list.tsx [hunk 2]
```diff
@@ -80,6 +82,15 @@ interface IBranchListProps {
 
   /** Callback to fire when the filter text is changed */
   readonly onFilterTextChanged: (filterText: string) => void
+
+  /** Should it let users create a new branch? */
+  readonly allowCreateNewBranch: boolean
```
Comment (shiftkey) [inline app/src/ui/branches/branch-list.tsx]: Could we stick to the `canXYZ` convention on props - e.g. `canSelectRow`, `canCommit` - here?

**Comment by joshaber** [comment]
🇧🇲

**Pull request merged**